execution/engine.go

```diff
@@ -64,19 +64,28 @@ type PullEngine struct {
 	storage                storage.Storage
 	streamID               *StreamID
 	batchSizeManager       *BatchSizeManager
+	shouldPrefixStreamID   bool
 }
 
-func NewPullEngine(irs IntermediateRecordStore, storage storage.Storage, source RecordStream, streamID *StreamID, watermarkSource WatermarkSource) *PullEngine {
+func NewPullEngine(irs IntermediateRecordStore, storage storage.Storage, source RecordStream, streamID *StreamID, watermarkSource WatermarkSource, shouldPrefixStreamID bool) *PullEngine {
 	return &PullEngine{
-		irs:              irs,
-		storage:          storage,
-		source:           source,
-		streamID:         streamID,
-		watermarkSource:  watermarkSource,
-		batchSizeManager: NewBatchSizeManager(time.Second / 4),
+		irs:                  irs,
+		storage:              storage,
+		source:               source,
+		streamID:             streamID,
+		watermarkSource:      watermarkSource,
+		batchSizeManager:     NewBatchSizeManager(time.Second / 4),
+		shouldPrefixStreamID: shouldPrefixStreamID,
```

Comment:
What was the idea behind this change? Why do some pull engines need to prefix their transactions with their streamID and some don't?

Reply:
Because usually it makes stuff easier for the intermediate record store beneath it.

However, stream senders and receivers need to access common queues, which are prefixed by the shuffle id, not stream id.